packages/mermaid/src/diagrams/sankey/sankeyTypes.ts

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import type { SankeyNode as d3SankeyNode, SankeyLink as d3SankeyLink } from 'd3-sankey';
+
+import type { SankeyDiagramConfig } from '../../config.type.js';
+import type { DiagramDB } from '../../diagram-api/types.js';
+
+export interface SankeyFields {
+  links: SankeyLink[];
+  nodes: string[];
```

Comment:
Should not nodes be unique? Does sankey-d3 work fine with non-unique dataset?

Reply:
The thing is that langium doesn't support `Set` type, that's why I used `string[]` instead, the parser would make them unique:

https://github.com/mermaid-js/mermaid/blob/b6906a6474e01b5a6946749700150a4bada61657/packages/parser/src/language/sankey/sankeyParser.ts#L14-L25

Although the user might modify the db.